lib/tezos/lib/snapshot-node-stack.ts

```diff
@@ -0,0 +1,212 @@
+                    "arn:aws:s3:::amazoncloudwatch-agent",
+                    "arn:aws:s3:::amazoncloudwatch-agent/*",
+                    "arn:aws:s3:::lambsonacid-octez-*"
+                ],
+                actions: ["s3:ListBucket", "s3:*Object", "s3:GetBucket*"],
+            })
+        );
+        const snapshotNodeScript = fs.readFileSync(path.join(__dirname, "assets", "user-data", "node.sh")).toString();
+
+
+        const snapshotNode = new ec2.Instance(this, "tz-snapshot-node", {
+            instanceName: "tz-snapshot-node",
+            instanceType: instanceType,
+            machineImage: new ec2.AmazonLinux2023ImageSsmParameter({
+                kernel: ec2.AmazonLinux2023Kernel.KERNEL_6_1,
+                cpuType: instanceCpuType,
+            }),
+            vpc: vpc,
+            blockDevices: [
+                {
+                  // ROOT VOLUME
+                  deviceName: "/dev/xvda",
+                  volume: ec2.BlockDeviceVolume.ebs(46, {
+                      deleteOnTermination: true,
+                      encrypted: true,
+                      iops: 3000,
+                      volumeType: ec2.EbsDeviceVolumeType.GP3,
+                    }),
+                },
+              ],
+            detailedMonitoring: true,
+            propagateTagsToVolumeOnCreation: true,
+            role: snapshotInstanceRole,
+            securityGroup: instanceSG.securityGroup,
+          });
+
+          const modifiedSnapshotNodeScript = cdk.Fn.sub(snapshotNodeScript, {
+              _AWS_REGION_: REGION,
+              _STACK_NAME_: STACK_NAME,
+              _TZ_SNAPSHOTS_URI_: snapshotsUrl,
+              _STACK_ID_: STACK_ID,
+              _TZ_HISTORY_MODE_: historyMode,
+              _TZ_DOWNLOAD_SNAPSHOT_ : String(downloadSnapshot),
+              _TZ_NETWORK_: tzNetwork,
+              _S3_SYNC_BUCKET_: snapshotsBucket.bucketName,
+              _NODE_CF_LOGICAL_ID_: snapshotNode.instance.logicalId,
+              _LIFECYCLE_HOOK_NAME_: lifecycleHookName,
+              _AUTOSCALING_GROUP_NAME_: autoScalingGroupName,
+              _ASSETS_S3_PATH_: `s3://${asset.s3BucketName}/${asset.s3ObjectKey}`,
+              _INSTANCE_TYPE_: "SNAPSHOT",
+          });
+
+          snapshotNode.addUserData(modifiedSnapshotNodeScript);
+
+
+        // Getting logical ID of the instance to send ready signal later once the instance is initialized
+        const snapshotNodeCfn = snapshotNode.node.defaultChild as ec2.CfnInstance;
+
+        // CloudFormation Config: wait for 15 min for the node to start
+        const creationPolicy: cdk.CfnCreationPolicy = {
+            resourceSignal: {
+            count: 1,
+            timeout: "PT90M",
+            },
+        };
+
+
+        snapshotNodeCfn.cfnOptions.creationPolicy = creationPolicy;
+
+        new cdk.CfnOutput(this, "SnapshotBucketName", {
+            value: snapshotsBucket.bucketName,
+            exportName: "SnapshotBucketName",
```

Comment:
Use more unique export name value for snapshot bucket. For example: `TezosSnapshotBucket`.

Reply:
✅